libpod/util.go

```diff
@@ -55,8 +55,14 @@ func WaitForFile(path string, chWait chan error, timeout time.Duration) (bool, e
 		if err := watcher.Add(filepath.Dir(path)); err == nil {
 			inotifyEvents = watcher.Events
 		}
-		defer watcher.Close()
-		defer watcher.Remove(filepath.Dir(path))
+		defer func() {
+			if err := watcher.Close(); err != nil {
+				logrus.Errorf("failed to close fsnotify watcher: %v", err)
+			}
+			if err := watcher.Remove(filepath.Dir(path)); err != nil {
+				logrus.Errorf("failed to remove fsnotify watcher: %v", err)
+			}
```

Comment:
looks this is broken `failed to remove fsnotify watcher: bad file descriptor`

From the doc
> Close removes all watches and closes the events channel. 

so we only have to call Close()